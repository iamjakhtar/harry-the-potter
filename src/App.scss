@import "styles/colors";

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: "Roboto", sans-serif;
  font-size: 16px;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: "Poppins", sans-serif;
    font-weight: 600;
    color: $primary-color;
  }

  h1 {
    text-align: center;
    font-size: 48px;
    margin: 20px 0;
  }
  
  p {
      font-weight: 400;
      color: $dark-text-color
  }

  a {
    color: $secondary-color;
    text-decoration: none;

    :hover {
        color: $accent-color;
    }
  }
}

