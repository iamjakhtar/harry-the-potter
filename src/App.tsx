import "./App.scss";
import BookList from "./components/book-list/BookList";
import Footer from "./components/footer/Footer";
import Header from "./components/header/Header";

function App() {
  

  return (
    <div className="App">
      <Header />
      <h1>Harry The Potter Collection</h1>
      <BookList />
      <Footer />
    </div>
  );
}

export default App;
