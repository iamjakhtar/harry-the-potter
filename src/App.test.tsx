import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import CartContextProvider from './context/CartContext';

test('renders without crashing', () => {
  const { container } = render(
    <CartContextProvider>
      <App />
    </CartContextProvider>
  );

  expect(container).toMatchSnapshot();
  
});
