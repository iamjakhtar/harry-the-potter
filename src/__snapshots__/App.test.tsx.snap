// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <nav>
        <a
          href="/"
        >
          Harry The Potter
        </a>
        <button>
          <img
            alt="Cart Icon"
            src="cart-shopping.svg"
          />
        </button>
      </nav>
    </header>
    <h1>
      Harry The Potter Collection
    </h1>
    <div
      class="book-list-container"
    >
      <div
        class="book-container"
      >
        <img
          alt="Harry The Potter and Harriet"
          src="book1.png"
        />
        <h3>
          Harry The Potter and Harr...
        </h3>
        <p>
          <span>
            Price:
          </span>
          <span>
            €
            8.00
          </span>
        </p>
        <button>
          add to cart
        </button>
      </div>
      <div
        class="book-container"
      >
        <img
          alt="Harry The Potter Hunger Games"
          src="book2.png"
        />
        <h3>
          Harry The Potter Hunger G...
        </h3>
        <p>
          <span>
            Price:
          </span>
          <span>
            €
            8.00
          </span>
        </p>
        <button>
          add to cart
        </button>
      </div>
      <div
        class="book-container"
      >
        <img
          alt="Harry The Potter and Blue Dragon"
          src="book3.png"
        />
        <h3>
          Harry The Potter and Blue...
        </h3>
        <p>
          <span>
            Price:
          </span>
          <span>
            €
            8.00
          </span>
        </p>
        <button>
          add to cart
        </button>
      </div>
      <div
        class="book-container"
      >
        <img
          alt="Harry The Potter and Black Sea"
          src="book4.png"
        />
        <h3>
          Harry The Potter and Blac...
        </h3>
        <p>
          <span>
            Price:
          </span>
          <span>
            €
            8.00
          </span>
        </p>
        <button>
          add to cart
        </button>
      </div>
      <div
        class="book-container"
      >
        <img
          alt="Harry The Potter and Talking Owl"
          src="book5.png"
        />
        <h3>
          Harry The Potter and Talk...
        </h3>
        <p>
          <span>
            Price:
          </span>
          <span>
            €
            8.00
          </span>
        </p>
        <button>
          add to cart
        </button>
      </div>
    </div>
    <div
      class="footer-container"
    >
      <p>
        © Harry The Potter
      </p>
    </div>
  </div>
</div>
`;
